### PR TITLE
Controllers specs

### PR DESCRIPTION
**Pull request by eapenzacharias**
- Added views for `Users` `#index` & `#show`
- Added views for `Posts` `#index` & `#show`
- Created tests for `Users` and `Posts` methods

### CODE REVIEW

File: test/controllers/posts_spec.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+RSpec.describe 'Posts', type: :request do
+  describe 'GET /index' do
+    before(:example) { get '/users/sample_user' }
+    it 'Should have http status ok' do
+      expect(response).to have_http_status(:ok)
+    end
+    it 'should render posts index page for a user' do
+      expect(response.body).to include('User: sample_user')
+    end
```
Comment (Emmyn5600) [inline test/controllers/posts_spec.rb]: - [ ] For this, ensure the response body includes the correct placeholder text. please target the correct placeholder that is appearing on the body.

File: test/controllers/posts_spec.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  end
+  describe 'GET /show' do
+    before(:example) { get '/users/sample_user/posts/post_id' }
+    it 'Should have http status ok' do
+      expect(response).to have_http_status(:ok)
+    end
+    it 'should render specific post page for given user' do
+      expect(response.body).to include('Post: post_id User: sample_user')
```
Comment (Emmyn5600) [inline test/controllers/posts_spec.rb]: - [ ] For this ensure the response body includes the correct placeholder text. please target the correct placeholder that is appearing on the body/.

File: test/controllers/users_spec.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    end
+  end
+  describe 'GET /show' do
+    before(:example) { get '/users/sample_user' }
+    it 'Should have http status ok' do
+      expect(response).to have_http_status(:ok)
+    end
+    it 'should render user index page' do
+      expect(response.body).to include('User: sample_user')
+    end
```
Comment (Emmyn5600) [inline test/controllers/users_spec.rb]: - [ ] For this ensure the response body includes the correct placeholder text. please target the correct placeholder that is appearing on the body.

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
Hi @eapenzacharias 

Good job so far! I know you have worked hard on making this milestone But  There are some issues that you still need to work on to go to the next project but you are almost there!

## To highlight :champagne:

- All linters are passing :heavy_check_mark:
- Good commit history :heavy_check_mark:
- Good PR description :heavy_check_mark:

## Required Changes :recycle:

- [ ] Check the comment under the review

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!:clap::clap::clap:

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: test/controllers/posts_spec.rb (hunk 1 above)
Comment (eapenzacharias) [reply test/controllers/posts_spec.rb]: Hello @Emmyn5600,
Thank you for your review. I didn't quite understand what you meant by these comments
Here I am checking for a text `'User: sample_user'` when I provide an id `sample_user`
Could you please elaborate on the changes I have to make

Comment (Emmyn5600) [reply test/controllers/posts_spec.rb]: @eapenzacharias what I meant was to target the correct placeholder text that is appearing on the body to ensure that we are getting the correct output after targeting that text.

Comment (eapenzacharias) [reply test/controllers/posts_spec.rb]: Hello @Emmyn5600

Thank you for the clarification. I believe in these three cases I am targeting the right placeholders. Could you please verify or am I missing something.

**Review by uwadonat — APPROVED** [review]
Hi **@[eapenzacharias](https://github.com/eapenzacharias),** 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️✔️✔️

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**